.github/workflows/conformance-gke.yaml

```diff
@@ -119,7 +118,44 @@ jobs:
 
           echo "Generated matrix:"
           cat /tmp/matrix.json
-          echo "matrix=$(jq -c . < /tmp/matrix.json)" >> $GITHUB_OUTPUT
+
+      - name: Set up gcloud credentials
+        id: 'auth'
+        uses: google-github-actions/auth@55bd3a7c6e2ae7cf1877fd1ccb9d54c0503c457c # v2.1.2
+        with:
+          workload_identity_provider: ${{ secrets.GCP_PR_WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ secrets.GCP_PR_SA }}
+          create_credentials_file: true
+          export_environment_variables: true
+
+      - name: Set up gcloud CLI
+        uses: google-github-actions/setup-gcloud@98ddc00a17442e89a24bbf282954a3b65ce6d200 # v2.1.0
+        with:
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          version: "405.0.0"
+
+      - name: Filter Matrix
+        id: set-matrix
+        run: |
+          cp /tmp/matrix.json /tmp/result.json
+          jq -c '.k8s[]' /tmp/matrix.json | while read i; do
+            VERSION=$(echo $i | jq -r '.version')
+            ZONE=$(echo $i | jq -r '.zone')
+            gcloud --quiet container get-server-config \
+              --flatten="channels" --filter="channels.channel=REGULAR"   \
+              --format="yaml(channels.validVersions)" --zone $ZONE > /tmp/output
+            if grep -q -F $VERSION /tmp/output; then
+              echo "Version $VERSION is valid for zone $ZONE"
+            else
+              echo "Removing version $VERSION as it's not valid for zone $ZONE"
```

Comment:
It could make sense to use [notice messages](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-setting-a-notice-message) to make this more evident. Example: https://github.com/cilium/cilium/actions/runs/8924272573 (displayed in the annotations pane).

Reply:
Good idea, done.